src/mina-signer/MinaSigner.ts

```diff
@@ -78,13 +109,16 @@ class Client {
    * @param key The key pair used to sign the message
    * @returns A signed message
    */
-  public signMessage(message: string, key: Keypair): Signed<Message> {
+  public signMessage(message: string, key: Json.Keypair): Signed<Json.Message> {
     return {
-      signature: minaSDK.signString(this.network, key.privateKey, message),
-      data: {
-        publicKey: key.publicKey,
-        message,
+      signature: {
+        ...signString(message, key.privateKey, this.network),
+        signer: key.publicKey,
       },
+      // TODO: returning the public key, which is an input, twice seems awfully redundant
+      // this should just be `data: message`
+      // and do we really need `signer` as part of `Signature`?
+      data: { publicKey: key.publicKey, message },
```

Comment:
@MartinMinkov what are the implications of making a breaking change here? We could deprécate this and introduce a new function I suppose?